sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkHttpClientProvider.java

```diff
@@ -13,9 +14,42 @@
  * introduced.
  */
 public final class JdkHttpClientProvider implements HttpClientProvider {
+    private static final boolean AZURE_DISABLE_HTTP_CLIENT_SHARING =
+        Configuration.getGlobalConfiguration().get("AZURE_DISABLE_HTTP_CLIENT_SHARING", Boolean.TRUE);
```

Comment:
I think we should default to false here and in the other place.

Reply:
Doesn't it break existing customers' behavior?

Reply:
The polarity of the switch might be backward. Since it is new behavior and existing behavior is the default, maybe it should be `AZURE_ENABLE_HTTP_CLIENT_SHARING`?

Reply:
Well that naming was chosen assuming that we roll out new default plus compat switch as an escape hatch if somebody would run into problems.

Reply:
Got it. Current name is great if we default to false to bring folks forward.

Reply:
I ended up running Shawn's jars on a VM (results in other comments on this PR) and it looks like there are mixed results. I've run two scenarios the `nonsharedhttpclientdownloadblob` and `downloadblob` to see how this impacts cases where client reference is kept at method scope vs application scope.
The outcome seems to be that:
- The change doesn't make any difference to Netty because it's sharing a connection pool anyway internally (with generous hundreds of connections).
- The OkHttp client fails miserably in method scope scenario - the JVM runs out of physical threads. Caused by OkHttp client spinning resources every time.
- The OkHttp underperforms if shared in app scope scenarios. Because the default connection pool is small, i.e. 5 connections.

Given that I think I'm gonna agree with @billwert and recommend to make it opt-in feature rather than flipping the default. Unless at least we address OkHttp small connection pool.

Reply:
updated. Thanks Bill and Kamil